src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/PackageSpecificWarningProperties.cs

```diff
@@ -88,24 +88,21 @@ public static PackageSpecificWarningProperties CreatePackageSpecificWarningPrope
         /// <param name="framework">Target graph for which no warning should be thrown.</param>
         public void Add(NuGetLogCode code, string libraryId, NuGetFramework framework)
         {
-            if (Properties == null)
-            {
-                Properties = new Dictionary<NuGetLogCode, IDictionary<string, ISet<NuGetFramework>>>();
-            }
+            Properties ??= new Dictionary<NuGetLogCode, IDictionary<string, ISet<NuGetFramework>>>();
 
-            if (!Properties.ContainsKey(code))
+            if (!Properties.TryGetValue(code, out IDictionary<string, ISet<NuGetFramework>> frameworksByLibraryId))
```

Comment:
looks like we have a bunch of duplication in these classes. I created a follow up, cause I bet there's more :D https://github.com/NuGet/Home/issues/12676